lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -165,48 +170,285 @@ def test_clear_tools_test69(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    assert pidstat3_tool.exists() is False
 
 
 def test_clear_tools_test70(monkeypatch, agent_config, pbench_run, pbench_cfg):
     """ Attempt to clear the wrong tool, by name """
     monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
     default_group = pbench_run / "tools-v1-default"
     default_group.mkdir(parents=True)
-    fubar5_host = default_group / "fubar5.example.com"
-    fubar5_host.mkdir(parents=True, exist_ok=True)
-    pidstat5_tool = fubar5_host / "pidstat"
-    pidstat5_tool.touch()
+    fubar_host = default_group / "fubar.example.com"
+    fubar_host.mkdir(parents=True, exist_ok=True)
+    pidstat_tool = fubar_host / "pidstat"
+    pidstat_tool.touch()
 
     command = ["pbench-clear-tools", "--name=vmstat"]
     out, err, exitcode = pytest.helpers.capture(command)
     assert b"" == out
-    assert b'Tool "vmstat" not found' in err
+    assert b"Tools ['vmstat'] not found in group default\n" in err
     assert exitcode == 0
+
+
+def test_clear_tools_test71(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear the tool groups, by name, separated by comma"""
+    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
+    default_group = pbench_run / "tools-v1-default"
+    default_group.mkdir(parents=True)
+    fubar_default_host = default_group / "fubar6.example.com"
+    fubar_default_host.mkdir(parents=True, exist_ok=True)
+    pidstat_default_tool = fubar_default_host / "pidstat"
+    pidstat_default_tool.touch()
+
+    another_group = pbench_run / "tools-v1-another"
+    another_group.mkdir(parents=True)
+    fubar_another_host = another_group / "fubar7.example.com"
+    fubar_another_host.mkdir(parents=True, exist_ok=True)
+    pidstat_another_tool = fubar_another_host / "pidstat"
+    pidstat_another_tool.touch()
+
+    command = ["pbench-clear-tools", "--groups=default,another"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert default_group.exists() is True
+    assert another_group.exists() is False
+
+
+def test_clear_tools_test72(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear the tool groups, by name, separated by comma.
+    However one of the tool group name is wrong"""
+    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
+
+    another_group = pbench_run / "tools-v1-another"
+    another_group.mkdir(parents=True)
+    fubar_another_host = another_group / "fubar.example.com"
+    fubar_another_host.mkdir(parents=True, exist_ok=True)
+    pidstat_another_tool = fubar_another_host / "pidstat"
+    pidstat_another_tool.touch()
+
+    command = ["pbench-clear-tools", "--groups=another,wrong"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert (
+        b'Removed "pidstat" from host "fubar.example.com" in tools group '
+        b'"another"' in err
+    )
+    assert b'pbench-clear-tools: invalid --group option "wrong"' in err
```

Comment:
I'm not even going to *say* this, so you didn't read it; but as I was glancing through checking for alternate uses of option aliases, I happened to notice that this test case uses `--groups` but the message says `--group`. Is that potentially confusing? Do we care? Is there any reasonable way we could improve that UX? (I.e., does `click` let us determine which alias was actually used?)

Remember, I didn't say this, and you didn't read it ...